state_mapper:
  ros__parameters:
    # Order in the following 2 lists is important for mapping
    roboy_link_names:
      [torso,
      upperarm_right,
      lowerarm_right,
      hand_right,
      upperarm_left,
      lowerarm_left,
      hand_left,
      head]
    operator_link_names:
      [chest,
      right_shoulder,
      right_elbow,
      right_wrist,
      left_shoulder,
      left_elbow,
      left_wrist,
      neck]

exoforce:
  ros__parameters:
    no_slack_force: 4.0       # (Newtons) Minimal force applied to the tendons to avoid slack
    tendon_init_time: 3.0     # (seconds) Time to wait after starting force controllers to pull the initial slack
    max_force_apply_time: 1.0 # (seconds) Max time the last decomposed force will be applied
    force_decomposition:
      min_tendon_force: 0.5
      max_collision_force: 50.0
    map_link_to_ef:
      left_wrist: [7, 8, 43]
      right_wrist: [4, 5, 3]

operator:
  ros__parameters:
    pull_motor: 16
    pull_force: 30.0
    pull_time: 1.0
    operator_link_dimentions:
      chest:
        [0.24991999999999998,
        0.24991999999999998,
        0.24991999999999992]
      left_elbow:
        [0.09288000000000529,
        0.0928800798163818,
        0.24624002969910885]
      left_shoulder:
        [0.1042400000000326,
        0.10424010023452646,
        0.30872003341150434]
      left_wrist:
        [0.09087999999999999,
        0.09087999999999996,
        0.09088000000000007]
      right_elbow:
        [0.09288000000000397,
        0.09288007981638124,
        0.24624002969910863]
      right_shoulder:
        [0.10424000000003777,
        0.10424010023452582,
        0.30872003341150656]
      right_wrist:
        [0.09087999999999999,
        0.09088000000000002,
        0.09088000000000007]
      neck:
        [0.23287999999999998,
        0.23287999999999998,
        0.23287999999999975]


# Unmapped Robot links

# wrist_right_link1
# wrist_right_link2
# wrist_right_link
# rh_palm
# wrist_left_link1
# wrist_left_link2
# wrist_left_link
# head_link2
# head_link1
# lh_palm
# elbow_right
# shoulder_right_link2
# shoulder_right_link1
# shoulder_right_link
# elbow_left
# shoulder_left_link1
# shoulder_left_link2
# shoulder_left_link
# camera_left
# camera_right
# lh_manipulator
# lh_ffknuckle
# lh_ffproximal
# lh_ffmiddle
# lh_ffdistal
# lh_fftip
# lh_mfknuckle
# lh_mfproximal
# lh_mfmiddle
# lh_mfdistal
# lh_mftip
# lh_rfknuckle
# lh_rfproximal
# rest of finger links

# Unmapped Human links

# base
# root
# neck
# right_hip
# righht_knee
# right_ankle
# left_hip
# left_knee
# left_ankle